commands/command_migrate.go

```diff
@@ -0,0 +1,230 @@
+		// set, respectively.
+		include = append(include, migrateIncludeRefs...)
+		exclude = append(exclude, migrateExcludeRefs...)
+	} else {
+		// Otherwise, if neither --include-ref=<ref> or
+		// --exclude-ref=<ref> were given, include no additional
+		// references, and exclude all remote references that are remote
+		// branches or remote tags.
+		remoteRefs, err := getRemoteRefs()
+		if err != nil {
+			return nil, nil, err
+		}
+
+		exclude = append(exclude, remoteRefs...)
+	}
+
+	return include, exclude, nil
+}
+
+// getRemoteRefs returns a fully qualified set of references belonging to all
+// remotes known by the currently checked-out repository, or an error if those
+// references could not be determined.
+func getRemoteRefs() ([]string, error) {
+	var refs []string
+
+	remotes, err := git.RemoteList()
+	if err != nil {
+		return nil, err
+	}
+
+	for _, remote := range remotes {
+		refsForRemote, err := git.RemoteRefs(remote)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, ref := range refsForRemote {
+			refs = append(refs, formatRefName(ref, remote))
+		}
+	}
+
+	return refs, nil
+}
+
+// formatRefName returns the fully-qualified name for the given Git reference
+// "ref".
+func formatRefName(ref *git.Ref, remote string) string {
+	var name []string
+
+	switch ref.Type {
+	case git.RefTypeRemoteBranch:
+		name = []string{"refs", "remotes", remote, ref.Name}
+	case git.RefTypeRemoteTag:
+		name = []string{"refs", "tags", ref.Name}
+	default:
+		return ref.Name
+	}
+	return strings.Join(name, "/")
+
+}
+
+// currentRefToMigrate returns the fully-qualified name of the currently
+// checked-out reference, or an error if the reference's type was not a local
+// branch.
+func currentRefToMigrate() (*git.Ref, error) {
+	current, err := git.CurrentRef()
+	if err != nil {
+		return nil, err
+	}
+
+	if current.Type == git.RefTypeOther ||
+		current.Type == git.RefTypeRemoteBranch ||
+		current.Type == git.RefTypeRemoteTag {
+
+		return nil, errors.Errorf("fatal: cannot migrate non-local ref: %s", current.Name)
+	}
+	return current, nil
+}
+
+// getHistoryRewriter returns a history rewriter that includes the filepath
+// filter given by the --include and --exclude arguments.
+func getHistoryRewriter(cmd *cobra.Command, db *odb.ObjectDatabase) *githistory.Rewriter {
+	include, exclude := getIncludeExcludeArgs(cmd)
+	filter := buildFilepathFilter(cfg, include, exclude)
+
+	return githistory.NewRewriter(db, githistory.WithFilter(filter))
+}
+
+func init() {
+	RegisterCommand("migrate", nil, func(cmd *cobra.Command) {
+		// Adding flags directly to cmd.Flags() doesn't apply those
+		// flags to any subcommands of the root. Therefore, loop through
+		// each subcommand specifically, and include common arguments to
+		// each.
+		//
+		// Once done, link each orphaned command to the
+		// `git-lfs-migrate(1)` command as a subcommand (child).
+
+		for _, subcommand := range []*cobra.Command{
+			MigrateInfoCommand,
```

Comment:
I don't think it makes sense to declare this `MigrateInfoCommand` var in `commands/command_migrate_info.go` if some of the flags are added here. They should all be set up in _this_ `init()` func. What do you think?